src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -183,33 +206,84 @@ const DotPlotPage = (props) => {
-      dispatch(fetchPlotDataWork(experimentId, plotUuid, plotType));
+
+      // dispatch is different depending on whether selected genes don't change or are added, re-ordered or deleted
+      // to prevent unnecessary rerenders
+      if (_.isEqual(currentSelected, previousSelected)) {
+        dispatch(fetchPlotDataWork(experimentId, plotUuid, plotType));
+        return;
+      }
+
+      if (currentSelected.length > previousSelected.length) {
+        dispatch(fetchPlotDataWork(experimentId, plotUuid, plotType));
+        setReorderFetched(true);
+        return;
+      }
+
+      if (currentSelected.length === previousSelected.length) {
+        reorderData(currentSelected);
+        return;
+      }
+      
+      // if (currentSelected.length < previousSelected.length) {
+      const removedGene = previousSelected.filter((gene) => !currentSelected.includes(gene))[0];
+      deleteData(removedGene);
+      // }
     }
   }, [config, cellSetProperties]);
 
-  const loadInitialCustomGenes = () => {
-    const PROPERTIES = ['dispersions'];
-    const tableState = {
+  // load the gene names and dispersions for search table and initial state
+  useEffect(() => {
+    const state = {
+      sorter: {
+        field: 'gene_names',
+        columnKey: 'gene_names',
+        order: 'ascend',
+      },
       pagination: {
-        current: 1, pageSize: config.nMarkerGenes, showSizeChanger: true, total: 0,
+        current: 1,
+        pageSize: 100000,
       },
-      geneNamesFilter: null,
-      sorter: { field: PROPERTIES[0], columnKey: PROPERTIES[0], order: 'descend' },
+      pageSizeFilter: null,
     };
 
-    dispatch(loadPaginatedGeneProperties(experimentId, PROPERTIES, plotUuid, tableState));
+    dispatch(loadPaginatedGeneProperties(experimentId, ['dispersions'], searchBarUuid, state));
+  }, []);
+
+  // find genes with highest dispersion from list of genes sorted by name
+  const loadHighestDispersionGenes = () => {
+    const highestDispersions = Object.values(geneData).map((gene) => gene.dispersions).sort().splice(-config.nMarkerGenes);
+
+    const getKeyByValue = (value) => Object.keys(geneData).find((key) => geneData[key].dispersions === value);
+
+    const highestDispersionGenes = highestDispersions.map((dispersion) => getKeyByValue(dispersion));
+
+    updatePlotWithChanges({ selectedGenes: highestDispersionGenes });
   };
 
+  // load initial state, based on highest dispersion genes from all genes
   useEffect(() => {
-    if (config?.selectedGenes.length === 0 && !highestDispersionGenes?.length > 0 && !genesFetching) {
-      loadInitialCustomGenes();
+    if (Object.keys(geneData).length === 0 || !config) {
+      return;
     }
 
-    if (config?.selectedGenes.length === 0 && highestDispersionGenes?.length > 0) {
-      updatePlotWithChanges({ selectedGenes: highestDispersionGenes });
-    }
-  }, [highestDispersionGenes, config, genesFetching]);
+    loadHighestDispersionGenes();
+  }, [geneData]);
+
```

Comment:
You use `config` here, but you don't have it in the dependency array on line 276. This means if config changes, this won't be recalculated. Is that what's expected?

Reply:
I removed the `config` from the if. This is meant to load the initial data to the plot (genes with the highest dispersion) after `geneData` is loaded (the list of all genes).